### ISSUE TITLE
Application Preferences

### ISSUE THREAD

**Issue by sabjorn** [sabjorn/BandcampEnhancementSuite#51]
It would be nice to be able to selective enable/disable features. This could then be extended for each feature to have some fine grained control (for example: configuring step size for jump forward/back audio feature; and/or changing the keybindings for the keyboard controls).

Ideally this would be a framework that is easily extended for each feature (each feature script talks to the background.js and acquires all the information it needs when it loads). Some sort of config struct could be used to unify the front/back end parameters.

Visually, nested menus would be good but to start off maybe just an omni list of controls?

**Comment by emcniece** [comment]
➕ 1 for simplicity to start. A settings page can take up as much room as it needs to.

I don't think proxying through `background.js` is necessary - if I recall, the typical pattern for this is setting a default settings object in localstorage then accessing it whenever needed, as localStorage is shared in the whole extension.

**Comment by sabjorn** [comment]
Isn't local storage is per domain?

**Comment by emcniece** [comment]
> Isn't local storage is per domain?

Yes, and I think the domain for your extension is something like chrome://extensions/?id=emgkbcgolkdoaaalacdocflfjccadjeo

**Comment by sabjorn** [comment]
@emcniece only for background scripts. So, yes, this could be implemented by having `local storage` connected to in a background script (like IDB is being used). However, I already have IDB stuff working and it doesn't have the storage limitations of `local storage`.

**Comment by sabjorn** [comment]
it is possible to have an "options" page which becomes a right-click selectable element on the extension icon:

https://developer.chrome.com/docs/extensions/mv3/getstarted/#options

**Comment by sabjorn** [comment]
https://developer.chrome.com/docs/extensions/mv3/options/

**Comment by sabjorn** [comment]
as part of this work we need to cleanup the 'config' backend -- and potentially creating a separate interface for config which is not handled by users. This way -- we can hopefully have the config page essentially auto-populate from the settings available in the config backend type definition

**Comment by sabjorn** [comment]
now setup and will be added to by any feature needing it
